independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/SubclassGenerator.java

```diff
@@ -138,9 +138,35 @@ protected FieldDescriptor createConstructor(ClassOutput classOutput, BeanInfo be
                         parameterTypes.subList(0, superParamsSize).toArray(new String[0])),
                 constructor.getThis(), superParams);
 
+        // we build a map for each interceptor instance created, so that they are shared
+        // Map<String, Object> where InjectableInterceptor.getIdentifier() is key and Object is instance of the interceptor for this bean
+        FieldCreator interceptorInstanceMap = subclass.getFieldCreator("interceptorInstanceMap", Map.class.getName())
+                .setModifiers(ACC_PRIVATE | ACC_FINAL);
+        // interceptorInstanceMap = new HashMap<>()
+        constructor.writeInstanceField(interceptorInstanceMap.getFieldDescriptor(), constructor.getThis(),
+                constructor.newInstance(MethodDescriptor.ofConstructor(HashMap.class)));
+        // create ResultHandleFor this Map
+        ResultHandle interceptorInstanceMapHandle = constructor.readInstanceField(interceptorInstanceMap.getFieldDescriptor(),
+                constructor.getThis());
+        // build a map that links InterceptorInfo to ResultHandle
         Map<InterceptorInfo, ResultHandle> interceptorToResultHandle = new HashMap<>();
         for (int j = 0; j < boundInterceptors.size(); j++) {
             interceptorToResultHandle.put(boundInterceptors.get(j), constructor.getMethodParam(j + superParamsSize + 1));
+
+            // create instance of each interceptor -> InjectableInterceptor.get()
+            ResultHandle creationalContext = constructor.invokeStaticMethod(MethodDescriptors.CREATIONAL_CTX_CHILD,
+                    creationalContextHandle);
+            ResultHandle interceptorInstance = constructor.invokeInterfaceMethod(
+                    MethodDescriptors.INJECTABLE_REF_PROVIDER_GET,
+                    constructor.getMethodParam(j + superParamsSize + 1), creationalContext);
```

Comment:
Maybe it's worth creating a `ResultHandle` for this one? It's used 3 times AFAICS.

Reply:
Good point, fixed!